Plugins/GRPCGeneratorPlugin/Plugin.swift

```diff
@@ -0,0 +1,270 @@
+          config: config,
+          protoDirectoryPath: protoDirectoryPath,
+          protocPath: protocPath,
+          protocGenSwiftPath: protocGenSwiftPath
+        )
+        commands.append(protoCommand)
+      }
+    }
+
+    return commands
+  }
+}
+
+/// Reads the configuration files at the supplied URLs into memory
+/// - Parameter configurationFiles: URLs from which to load configuration
+/// - Returns: A map of source URLs to loaded configuration
+func readConfigurationFiles(
+  _ configurationFiles: [URL],
+  pluginWorkDirectory: URL
+) throws -> [URL: CommonConfiguration] {
+  var configs: [URL: CommonConfiguration] = [:]
+  for configFile in configurationFiles {
+    let data = try Data(contentsOf: configFile)
+    let configuration = try JSONDecoder().decode(ConfigurationFile.self, from: data)
+
+    var config = CommonConfiguration(configurationFile: configuration)
+    // hard-code full-path to avoid collisions since this goes into a temporary directory anyway
+    config.fileNaming = .fullPath
+    // the output directory mandated by the plugin system
+    config.outputPath = String(pluginWorkDirectory.relativePath)
+    configs[configFile] = config
+  }
+  return configs
+}
+
+/// Finds the most precisely relevant config file for a given proto file URL.
+/// - Parameters:
+///   - file: The path to the proto file to be matched.
+///   - configFiles: The paths to all known configuration files.
+/// - Returns: The path to the most precisely relevant config file if one is found, otherwise `nil`.
+func findApplicableConfigFor(file: URL, from configFiles: [URL]) -> URL? {
+  let filePathComponents = file.pathComponents
+  for endComponent in (0 ..< filePathComponents.count).reversed() {
+    for configFile in configFiles {
+      if filePathComponents[..<endComponent]
+        == configFile.pathComponents[..<(configFile.pathComponents.count - 1)]
+      {
+        return configFile
+      }
+    }
+  }
+
+  return nil
+}
+
+/// Construct the command to invoke `protoc` with the `proto-gen-grpc-swift` plugin.
+/// - Parameters:
+///   - inputFile: The input `.proto` file.
+///   - configFile: The path file containing configuration for this operation.
+///   - config: The configuration for this operation.
+///   - protoDirectoryPath: The root path to the source `.proto` files used as the reference for relative path naming schemes.
+///   - protocPath: The path to `protoc`
+///   - protocGenGRPCSwiftPath: The path to `proto-gen-grpc-swift`.
+/// - Returns: The command to invoke `protoc` with the `proto-gen-grpc-swift` plugin.
+func protocGenGRPCSwiftCommand(
+  inputFile: URL,
+  configFile: URL,
+  config: CommonConfiguration,
+  protoDirectoryPath: URL,
+  protocPath: URL,
+  protocGenGRPCSwiftPath: URL
+) throws -> PackagePlugin.Command {
+  guard let fileNaming = config.fileNaming else {
+    assertionFailure("Missing file naming strategy - should be hard-coded.")
+    throw PluginError.missingFileNamingStrategy
+  }
```

Comment:
IDK if an error is appropriate here; the user literally can't do anything about it. I think we should `precondition` that the file naming is full path here. If it isn't then we've made a mistake.

Reply:
With the config restructuring this doesn't exist anymore.